### PR TITLE
feat: add support for base and base sepolia

### PR DESCRIPTION
**Pull request by spsjvc**
Test transaction: https://sepolia.basescan.org/tx/0x468759687eb73f9a59dba020a430fd921694bc382343ae43e3f62d69f3133156

Closes FS-523.

### CODE REVIEW

File: src/__snapshots__/getDefaultSequencerInboxMaxTimeVariation.unit.test.ts.snap [hunk 1]
```diff
@@ -50,5 +50,17 @@ exports[`returns default value for sequencerInboxMaxTimeVariation based on paren
     "futureBlocks": 300n,
     "futureSeconds": 3600n,
   },
+  "8453": {
```
Comment (spsjvc) [inline src/__snapshots__/getDefaultSequencerInboxMaxTimeVariation.unit.test.ts.snap:53]: x6 the usual number of blocks

File: src/__snapshots__/getDefaultConfirmPeriodBlocks.unit.test.ts.snap [hunk 2]
```diff
@@ -10,5 +10,7 @@ exports[`returns default value for confirmPeriodBlocks based on parent chain 1`]
   "42161": 45818n,
   "421614": 150n,
   "42170": 45818n,
+  "8453": 274908n,
```
Comment (spsjvc) [inline src/__snapshots__/getDefaultConfirmPeriodBlocks.unit.test.ts.snap:13]: x6 the usual number of blocks

File: src/__snapshots__/getDefaultConfirmPeriodBlocks.unit.test.ts.snap [hunk 3]
```diff
@@ -10,5 +10,7 @@ exports[`returns default value for confirmPeriodBlocks based on parent chain 1`]
...
+  "84532": 900n,
```
Comment (spsjvc) [inline src/__snapshots__/getDefaultConfirmPeriodBlocks.unit.test.ts.snap:14]: x6 the standard number of blocks, seconds stay the same

File: src/__snapshots__/getDefaultSequencerInboxMaxTimeVariation.unit.test.ts.snap [hunk 4]
```diff
@@ -50,5 +50,17 @@ exports[`returns default value for sequencerInboxMaxTimeVariation based on paren
...
+    "delaySeconds": 345600n,
+    "futureBlocks": 1800n,
+    "futureSeconds": 3600n,
+  },
+  "84532": {
+    "delayBlocks": 172800n,
+    "delaySeconds": 345600n,
+    "futureBlocks": 1800n,
+    "futureSeconds": 3600n,
+  },
```
Comment (spsjvc) [inline src/__snapshots__/getDefaultSequencerInboxMaxTimeVariation.unit.test.ts.snap:64]: x6 the standard number of blocks, seconds stay the same

File: src/getDefaultSequencerInboxMaxTimeVariation.ts [hunk 5]
```diff
@@ -12,17 +13,25 @@ export type SequencerInboxMaxTimeVariation = {
...
+  const parentChainId = validateParentChain(parentChainIdOrPublicClient);
 
   const delaySeconds = 60 * 60 * 24 * 4; // 4 days;
-  const delayBlocks = delaySeconds / 12;
-
   const futureSeconds = 60 * 60; // 1 hour;
-  const futureBlocks = futureSeconds / 12;
+
+  if (parentChainId === base.id || parentChainId === baseSepolia.id) {
+    // For Arbitrum L3s built on top of an OP Stack L2, `block.number` will return the L2 block number.
```
Comment (TucksonDev) [inline src/getDefaultSequencerInboxMaxTimeVariation.ts:22]:
```suggestion
    // For Orbit L3s built on top of an OP Stack L2, `block.number` will return the L2 block number.
```
Not sure whether to use Orbit or Arbitrum here (not my call either, lol)

Comment (TucksonDev) [reply src/getDefaultSequencerInboxMaxTimeVariation.ts:22]: (If you apply this change, the same comment appears in getDefaultConfirmPeriodBlocks.

**Review by TucksonDev — APPROVED** [review]
LGTM 🚀 
Thanks sir!

**Pull request merged**